ERCS/erc-7829.md

```diff
@@ -0,0 +1,135 @@
+author: Allen Dong (@Allen2730), Lonika Zhang <lonika@memolabs.net>, Steven He <steven@memolabs.net>
+discussions-to: https://ethereum-magicians.org/t/erc-7829-data-asset-nft/21881
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-11-29
+requires: 165, 721
+---
+
+## Abstract
+
+This proposal is an extension of [ERC-721](./eip-721.md). This proposal introduces a new role, Reader, who is granted by the Owner and allows multiple Readers for a single NFT. Additionally, this proposal extends the metadata interface of ERC-721, requiring that metadata should include at least a commitment, size, expire, and uploader's address. This proposal also proposes a storage proof mechanism to ensure the correctness of metadata information.
+
+## Motivation
+
+ERC-721 proposed the NFTs to represent the ownership of digital or physical assets. Currently, the NFT metadata is considered the NFT content, and its scarcity determines the value of the NFT. NFT owners can convert the content value of NFTs into revenue by transferring the ownership of NFTs. However, due to the high transaction fees, storage costs, and other expenses, NFTs are currently only able to represent the ownership of high-value assets, which limits the range of assets that NFTs can represent. This is especially true for data assets, where the content value of data assets varies greatly.
+
+Furthermore, to reduce storage costs, the NFT content is generally stored off-chain (IPFS, S3) or cross-chain storage (Arweave), and the link of NFT content is saved on-chain. Off-chain users can access the NFT content by visiting the link. However, on-chain contracts can not access the link to determine the status of the data, such as data loss, data tampering, or data expiration. These situations can lead to the data deviating from its actual value, but the NFT still exists on the chain, and the underlying copyright is still being sold in the market.
+
+This proposal introduces Data Asset NFTs, which solve the dilemma of on-chain data assets by combining a modular data layer--DA (Data Availability).
+
+## Specification
+
+### Terms
+
+In this proposal, we divide data assets into three parts:
+
+- Storage Metadata: Includes commitment, size, expire, and uploader's address. Stored and maintained by storage contracts on the blockchain.
+- Permission Metadata: Includes information on ownership and reading rights, as well as which addresses can modify this information. Stored and maintained by permission contracts on the blockchain.
+- Data Content: Data uploaded by users to the storage system. Data content is stored in off-chain storage nodes.
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY” and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### Every compliant data permission contract must implement the Interface
```

Comment:
Try to use shorter headings, and put the requirement in the body of the section.